Panic: nil pointer dereference in archivista client during verify

## Description

`witness verify` panics with nil pointer dereference when loading policy from local file without archivista configuration.

## Steps to Reproduce

```bash
witness verify -k pubkey.pem -p policy-signed.json -a attestation.att -f artifact.txt
```

## Error

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x10 pc=0x100542f58]

goroutine 1 [running]:
github.com/in-toto/go-witness/archivista.(*Client).archivistaRequestOpts(...)
	github.com/in-toto/go-witness@v0.9.1/archivista/client.go:56
github.com/in-toto/go-witness/archivista.(*Client).Download(...)
	github.com/in-toto/go-witness@v0.9.1/archivista/download.go:25 +0x38
github.com/in-toto/witness/internal/policy.LoadPolicy(...)
	github.com/in-toto/witness/internal/policy/policy.go:47 +0x288
```

## Environment

- go-witness: v0.9.1
- OS: macOS (Darwin 24.6.0)

## Expected Behavior

Verification should succeed or fail gracefully without panic when archivista is not configured.

## Workaround

Using the actual artifact file (instead of unrelated file) in `-f` flag avoids the panic.

--- a/archivista/client.go
+++ b/archivista/client.go
@@ -15,9 +15,15 @@
 package archivista
 
 import (
+	"errors"
 	"net/http"
 
 	"github.com/in-toto/archivista/pkg/api"
+)
+
+var (
+	ErrClientNotConfigured = errors.New("archivista client is not configured")
+	ErrURLNotConfigured    = errors.New("archivista URL is not configured")
 )
 
 type Client struct {
@@ -53,9 +59,22 @@ func New(url string, opts ...Option) *Client {
 
 func (c *Client) archivistaRequestOpts() []api.RequestOption {
 	opts := make([]api.RequestOption, 0)
-	if c.headers != nil {
-		opts = append(opts, api.WithHeaders(c.headers))
+	if c == nil || c.headers == nil {
+		return opts
 	}
 
+	opts = append(opts, api.WithHeaders(c.headers))
 	return opts
+}
+
+func (c *Client) validate() error {
+	if c == nil {
+		return ErrClientNotConfigured
+	}
+
+	if c.url == "" {
+		return ErrURLNotConfigured
+	}
+
+	return nil
 }

--- a/archivista/client_test.go
+++ b/archivista/client_test.go
@@ -1,0 +1,49 @@
+// Copyright 2022 The Witness Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package archivista
+
+import (
+	"context"
+	"testing"
+
+	"github.com/in-toto/go-witness/dsse"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClientMethodsWithNilReceiver(t *testing.T) {
+	var c *Client
+
+	_, err := c.Download(context.Background(), "sha256:deadbeef")
+	require.ErrorIs(t, err, ErrClientNotConfigured)
+
+	_, err = c.SearchGitoids(context.Background(), SearchGitoidVariables{})
+	require.ErrorIs(t, err, ErrClientNotConfigured)
+
+	_, err = c.Store(context.Background(), dsse.Envelope{})
+	require.ErrorIs(t, err, ErrClientNotConfigured)
+}
+
+func TestClientMethodsWithMissingURL(t *testing.T) {
+	c := New("")
+
+	_, err := c.Download(context.Background(), "sha256:deadbeef")
+	require.ErrorIs(t, err, ErrURLNotConfigured)
+
+	_, err = c.SearchGitoids(context.Background(), SearchGitoidVariables{})
+	require.ErrorIs(t, err, ErrURLNotConfigured)
+
+	_, err = c.Store(context.Background(), dsse.Envelope{})
+	require.ErrorIs(t, err, ErrURLNotConfigured)
+}

--- a/archivista/client_test.go
+++ b/archivista/client_test.go
@@ -1,4 +1,4 @@
-// Copyright 2022 The Witness Contributors
+// Copyright 2026 The Witness Contributors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.

--- a/archivista/download.go
+++ b/archivista/download.go
@@ -22,5 +22,9 @@ import (
 )
 
 func (c *Client) Download(ctx context.Context, gitoid string) (dsse.Envelope, error) {
+	if err := c.validate(); err != nil {
+		return dsse.Envelope{}, err
+	}
+
 	return archivistaapi.Download(ctx, c.url, gitoid, c.archivistaRequestOpts()...)
 }

--- a/archivista/searchgitoids.go
+++ b/archivista/searchgitoids.go
@@ -38,6 +38,10 @@ type SearchGitoidVariables struct {
 }
 
 func (c *Client) SearchGitoids(ctx context.Context, vars SearchGitoidVariables) ([]string, error) {
+	if err := c.validate(); err != nil {
+		return nil, err
+	}
+
 	const query = `query ($subjectDigests: [String!], $attestations: [String!], $collectionName: String!, $excludeGitoids: [String!]) {
   dsses(
     where: {

--- a/archivista/store.go
+++ b/archivista/store.go
@@ -22,6 +22,10 @@ import (
 )
 
 func (c *Client) Store(ctx context.Context, env dsse.Envelope) (string, error) {
+	if err := c.validate(); err != nil {
+		return "", err
+	}
+
 	resp, err := archivistaapi.Store(ctx, c.url, env, c.archivistaRequestOpts()...)
 	if err != nil {
 		return "", err

